modules/ROOT/pages/parse-template-reference.adoc

```diff
@@ -70,16 +70,69 @@ include::{examplesdir}/responseHtml.template[]
+To use the application in this example, you must send an HTTP request with a URL that includes an ID query parameter, such as `+http://localhost:8081/getEmployee?id=12345+`.
 
-==== Special Characters
+== Special Characters
 
-Parse Template supports the use of nested expression and literals inside expressions. There are some characters for special uses:
+Parse Template supports the use of expressions within a template and the use of literals within these expressions. You need to escape special characters of literal strings that are embedded in an expression.
 
-* An expression, which begins with the sequence `+#[+`, contains nested expressions. Inside an expression, the character `[` defines the start of an expression. The character  `[` does not need a preceding `+#+` . The character `[` is treated as a special character inside an expression.
-* Inside an expression, single quotes are used both to quote text and to treat text as a literal. So, the expression `+#['[']+` is a valid expression that evaluates to `[`. Outside expressions, single quotes are used as literals.
-* To avoid using a single quote to quote text inside an expression, add a backslash to escape it. For example `+#['this isn\'t an expression']+`.
-* Outside an expression, it is not necessary to quote characters `#` and `[` unless they are together. For example to create an expression just write `+\#[ the-expression-to-resolve ]+`.
+* `#[an-expression]`:
++
+Within a template, use the characters `+#[]+` to designate an expression. For example, a template can contain expressions within HTML elements, such as `<td>#[upper("mulesoft")]</td>`, which resolves to `<td>MULESOFT</td>`. Empty expressions (`+#[]+`) return an error, but empty strings, such as `+#['']+` or `+#[""]+` are valid expressions.
+
+* Sub-expressions:
++
+To resolve a sub-expression that is surrounded by quotation marks, you must surround the sub-expression with another `+#[]+`. For example, the three elements in the template snippet below contain the same sub-expression (`upper("world")`), either inside or outside of quotation marks. The top two return the same results. The bottom two do not process `upper("world")` because the sub-expression is inside the quotation marks but not surrounded by another `+#[]+`.
++
+.Parse Template Snippet:
+[souce,XML,linenums]
+----
+<td>#['hello #[upper("world")]']</td>
+<td>#['hello ' ++ upper("world")]</td>
+<td>#['hello upper("world")']</td>
+<td>#['hello ++ upper("world")']</td>
+----
++
+.Output Values:
+[souce,XML,linenums]
+----
+<td>hello WORLD</td>
+<td>hello WORLD</td>
+<td>hello upper("world")</td>
+<td>hello ++ upper("world")</td>
+----
+
+* Escape character (`\`):
++
+Parse Template uses the character sequence `\#[` to identify where an expression begins. To avoid this behavior and treat that sequence as literal characters, you must escape it with `\`. For example, `<td>\#[</td>` returns `<td>#[</td>`.
```

Comment:
note that the doc won't contain the backslash you see in the "`\#[`". Just using the backslash in the adoc file to escape the # so that it will show up in the HTML. The hash here won't show up in the HTML, only in this adoc file.